Add known issue for 8.15.x memory increase. (#1403)
elastic/ingest-docs: docs/en/ingest-management/release-notes/release-notes-8.15.asciidoc

[discrete]
[[known-issues-8.15.3]]
=== Known issues

[[known-issue-issue-41355-8.15.3]]
.The memory usage of {beats} based integrations is not correctly limited by the number of events actively in the memory queue, but rather the maximum size of the memory queue regardless of usage.
[%collapsible]
====

*Details*

In 8.15, events in the memory queue are not freed when they are acknowledged (as intended), but only when they are overwritten by later events in the queue buffer. This means for example if a configuration has a queue size of 5000, but the input data is low-volume and only 100 events are active at once, then the queue will gradually store more events until reaching 5000 in memory at once, then start replacing those with new events.

See {beats} issue link:https://github.com/elastic/beats/issues/40705[#40705].

*Impact* +

Memory usage may be higher than in previous releases depending on the throughput of {agent}. A fix is planned for 8.15.4.

- The worst memory increase is for low-throughput configs with large queues.
- For users whose queues were already sized proportionate to their throughput, memory use is increased but only marginally.

> - User that are affected can mitigate it by lowering their queue size.

Author: ```suggestion
- Affected users can mitigate the higher memory usage by lowering their queue size.
```

[discrete]
[[known-issues-8.15.2]]
=== Known issues

[[known-issue-issue-41355-8.15.2]]
.The memory usage of {beats} based integrations is not correctly limited by the number of events actively in the memory queue, but rather the maximum size of the memory queue regardless of usage.
[%collapsible]
====

*Details*

In 8.15, events in the memory queue are not freed when they are acknowledged (as intended), but only when they are overwritten by later events in the queue buffer. This means for example if a configuration has a queue size of 5000, but the input data is low-volume and only 100 events are active at once, then the queue will gradually store more events until reaching 5000 in memory at once, then start replacing those with new events.

See {beats} issue link:https://github.com/elastic/beats/issues/40705[#40705].

*Impact* +

Memory usage may be higher than in previous releases depending on the throughput of {agent}. A fix is planned for 8.15.4.

- The worst memory increase is for low-throughput configs with large queues.
- For users whose queues were already sized proportionate to their throughput, memory use is increased but only marginally.

> - User that are affected can mitigate it by lowering their queue size.

Reviewer: ```suggestion
- Affected users can mitigate the higher memory usage by lowering their queue size.
```

====

[[known-issue-issue-41355-8.15.1]]
.The memory usage of {beats} based integrations is not correctly limited by the number of events actively in the memory queue, but rather the maximum size of the memory queue regardless of usage.
[%collapsible]
====

*Details*

In 8.15, events in the memory queue are not freed when they are acknowledged (as intended), but only when they are overwritten by later events in the queue buffer. This means for example if a configuration has a queue size of 5000, but the input data is low-volume and only 100 events are active at once, then the queue will gradually store more events until reaching 5000 in memory at once, then start replacing those with new events.

See {beats} issue link:https://github.com/elastic/beats/issues/40705[#40705].

*Impact* +

Memory usage may be higher than in previous releases depending on the throughput of {agent}. A fix is planned for 8.15.4.

- The worst memory increase is for low-throughput configs with large queues.
- For users whose queues were already sized proportionate to their throughput, memory use is increased but only marginally.

> - User that are affected can mitigate it by lowering their queue size.

Reviewer: ```suggestion
- Affected users can mitigate the higher memory usage by lowering their queue size.
```

====

[[known-issue-issue-41355]]
.The memory usage of {beats} based integrations is not correctly limited by the number of events actively in the memory queue, but rather the maximum size of the memory queue regardless of usage.
[%collapsible]
====

*Details*

In 8.15, events in the memory queue are not freed when they are acknowledged (as intended), but only when they are overwritten by later events in the queue buffer. This means for example if a configuration has a queue size of 5000, but the input data is low-volume and only 100 events are active at once, then the queue will gradually store more events until reaching 5000 in memory at once, then start replacing those with new events.

See {beats} issue link:https://github.com/elastic/beats/issues/40705[#40705].

*Impact* +

Memory usage may be higher than in previous releases depending on the throughput of {agent}. A fix is planned for 8.15.4.

- The worst memory increase is for low-throughput configs with large queues.
- For users whose queues were already sized proportionate to their throughput, memory use is increased but only marginally.

> - User that are affected can mitigate it by lowering their queue size.

Reviewer: ```suggestion
- Affected users can mitigate the higher memory usage by lowering their queue size.
```
